pkg/tests/tasks/migration/migration_test.go

```diff
@@ -173,23 +315,113 @@ spec:
 	})
 }
 
-// Will continually request the URL until the context is cancelled and assert for success.
-func continuallyRequest(ctx context.Context, t test.TestHelper, url string) {
+// Will continually request the URL until the test has ended and assert for success.
+// Once the test is over, this func will clean itself up and wait until in flight
+// requests have finished.
+func continuallyRequest(t test.TestHelper, url string) {
 	t.T().Helper()
+	t.Logf("Continually requesting URL: %s", url)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	// 1. cancel
+	// 2. wait for in flight requests to finish so that the curl assertion doesn't fail the test if underlying resources have been deleted.
+	// 3. continue with other cleanups like deletion of resources.
+	stopped := make(chan struct{})
+	t.Cleanup(func() {
+		cancel()
+		t.Log("Waiting for continual requests to stop...")
+		<-stopped
+		t.Log("Continual requests stopped.")
+	})
 	go func(ctx context.Context) {
+	ReqLoop:
 		for {
+			if t.Failed() {
+				t.Log("Ending continual requests. Test failed.")
+				break
+			}
+
 			select {
 			case <-ctx.Done():
 				t.Log("Ending continual requests. Context has been cancelled.")
-				return
+				break ReqLoop
 			case <-time.After(time.Second):
-				curl.Request(t, url, curl.WithContext(ctx), assert.RequestSucceeds("productpage request succeeded", "productpage request failed"))
-			default:
-				if t.Failed() {
-					t.Log("Ending continual requests. Test failed.")
-					return
-				}
+				curl.Request(t, url, curl.WithContext(ctx), assert.RequestSucceedsAndIgnoreContextCancelled("productpage request succeeded", "productpage request failed"))
+				break ReqLoop
 			}
 		}
+		stopped <- struct{}{}
 	}(ctx)
 }
+
+// Detecting if the cluster supports ipv6 by looking for the kubernetes Service,
+// which should always be there, and inspecting ipFamilies.
+func clusterSupportsIPv6(t test.TestHelper) bool {
+	t.T().Helper()
+	var ipFamilies []string
+	ipFamResp := oc.GetJson(t, ns.Default, "Service", "kubernetes", "{.spec.ipFamilies}")
+	if err := json.Unmarshal([]byte(ipFamResp), &ipFamilies); err != nil {
+		t.Fatalf("Unable to marshal ip family resp: %s", err)
+	}
+
+	for _, ipFamily := range ipFamilies {
+		if ipFamily == "IPv6" {
+			return true
+		}
+	}
+
+	return false
+}
+
+func setupIstio(t test.TestHelper, istio ossm.Istio) {
+	t.T().Helper()
+	t.Cleanup(func() {
+		oc.DeleteResource(t, "", "Istio", istio.Name)
+		oc.DeleteResource(t, "", "IstioCNI", "default")
+	})
+	istioTempl := `apiVersion: sailoperator.io/v1alpha1
+kind: Istio
+metadata:
+  name: {{ .Name }}
+spec:
+  namespace: {{ .Namespace }}
+  version: {{ .Version }}`
```

Comment:
Just a thought. Do we need to specify the version explicitly for istio and istioCNI? (when no specified, operator picks the latest supported one)
Or do we want to have this options for migration pipeline to be able to install different istio version with the same operator?
cc: @FilipB  ^

Reply:
I confirmed with Marko, that this behavior is expected and I don't think we will ever want to install some older istio version for the migration test. So +1 not to specify version at all.